pkg/pdutil/placement_client.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pdutil
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"path"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/store/pdtypes"
+	"github.com/pingcap/tiflow/pkg/config"
+	"github.com/pingcap/tiflow/pkg/httputil"
+	"github.com/pingcap/tiflow/pkg/retry"
+	pd "github.com/tikv/pd/client"
+	"go.uber.org/zap"
+)
+
+const (
+	updateTypeDel = iota
+	updateTypeAdd
+)
+
+const (
+	// DefaultCheckInterval defines the default time interval for checking placement rules.
+	DefaultCheckInterval = 20 * time.Second
+
+	batchUpdatePrefix = "/pd/api/v1/config/rules/batch"
+	idQueryPrefix     = "/pd/api/v1/config/rule"
+
+	// `index` refers to the stacking sequence of rules in a group.
+	// `override` specifies whether to overwrite rules with smaller index (in a group).
+	addMetaJSON = `[{
+		"group_id": "pd",
+		"id": "cdcmeta",
+		"index": 20000,    
+		"override": true,  
+		"start_key": "",
+		"end_key": "6e00000000000000f8",
+		"role": "voter",
+		"count": 3,
```

Comment:
Does the `count` override other rules? eg. a rule that set count 5 for meta data.

Reply:
This rule will only be overridden if it is also configured under the `pd` group and index is less than cdcmeta.

Reply:
<img width="1259" alt="image" src="https://user-images.githubusercontent.com/61726649/160049884-7865d77f-1f98-42b3-8655-6920b332ef62.png">

Reply:
<img width="1212" alt="image" src="https://user-images.githubusercontent.com/61726649/160050225-fe41face-af56-4183-aa32-aaf1a6f96e66.png">

Reply:
@nolouch PTAL, what do you think about the `cdcmeta` rule? Do you have any suggestion?